### PR TITLE
[fix] (rpc-udf) Fixed the problem that the query could not be interrupted if an error occurred in the rpc server during the execution of rpc-udf.

### PR DESCRIPTION
**Pull request by yangzhg**
# Proposed changes

1. Fixed the problem that the query could not be interrupted
2. if an error occurred in the rpc server during the execution of rpc-udf.
Add `java`,`cpp`,`python` demo of rpc-udf server

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
compile failed

File: be/src/service/function_service.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+namespace brpc {
+class Controller;
+}
+
+namespace doris {
+
+class ExecEnv;
+
+class PFunctionServiceImpl : public PFunctionService {
```
Comment (morningman) [inline be/src/service/function_service.h:34]: What is this class for? A demo?

File: docs/zh-CN/extending-doris/udf/remote-user-defined-function.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Remote User Defined Function Service
+# 远程UDF
```
Comment (morningman) [inline docs/zh-CN/extending-doris/udf/remote-user-defined-function.md:27]: 可以在这个文章中列出 demo 的代码位置，以方便用户查看。
不然用户不太可能去 doris-demo 文档中查找。

File: be/src/service/function_service.h (hunk 1 above)
Comment (yangzhg) [reply be/src/service/function_service.h:34]: add by mistake ...  I will remove it

File: be/src/runtime/fragment_mgr.cpp [hunk 3]
```diff
@@ -250,15 +247,18 @@ Status FragmentExecState::cancel_before_execute() {
...
-    RETURN_IF_ERROR(_exec_status);
-    if (reason == PPlanFragmentCancelReason::LIMIT_REACH) {
-        _executor.set_is_report_on_cancel(false);
-    }
-    _executor.cancel();
-    if (_pipe != nullptr) {
-        _pipe->cancel(PPlanFragmentCancelReason_Name(reason));
+Status FragmentExecState::cancel(const PPlanFragmentCancelReason& reason, const std::string& msg) {
+    if (!_cancelled) {
+        _cancelled = true;
```
[the remark this answers is not in the document]
Comment (yangzhg) [reply be/src/runtime/fragment_mgr.cpp:252]: this is an bool type, and only set to true, so I think it is no need to use a lock

**Review by morningman — APPROVED** [review]
LGTM

**Comment by yangzhg** [comment]
Merger it for working on UDAF

**Pull request merged**